ext/spl/spl_directory.c

```diff
@@ -212,12 +212,22 @@ static inline void spl_filesystem_object_get_file_name(spl_filesystem_object *in
 			}
 			break;
 		case SPL_FS_DIR:
-			if (intern->file_name) {
-				efree(intern->file_name);
+			{
+				size_t path_len = 0;
+ 				char *path = spl_filesystem_object_get_path(intern, &path_len);
+				if (intern->file_name) {
+					efree(intern->file_name);
+				}
+				/* if there is parent path, ammend it, otherwise just use the given path as is */
+				if(path_len==0 && php_stream_is(intern->u.dir.dirp ,&php_glob_stream_ops)){
```

Comment:
Why is it necessary to special-case this code for the `glob` stream wrapper? Does something break if this special-casing is not done?

Reply:
If the special-casing is necessary (I'm not saying it isn't, just don't quite understand why), then this probably needs a `HAVE_GLOB` guard.

Reply:
Originally I only considered  check for path_len only a fix but the bug have a patch by Adam Harvey including a check for the `glob` stream wrapper, so I just wanted to be more careful.

But after your comment I spent some time poking around to check if it is actually breaking anything, I did not spot anything

so It makes sense for me to remove it,
also, I included the test cases for the behaviour changes to the PR

Reply:
@nikic I just fixed my tests for appveyor, was thinking that appveyor was timing out again 
it this PR okay now or need any thing else ?